submariner-operator/submariner-metrics-redesign.md

```diff
@@ -0,0 +1,163 @@
+modified to use `32781` instead of `8081`. A new env variable will be used for these ports.
+* [Optional] Modify `subctl cloud prepare` to no longer open metrics ports.
+
+### Submariner Metrics DaemonSet
+
+```yaml
+---
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: submariner-metrics-proxy
+spec:
+  selector:
+    matchLabels:
+      app: submariner-metrics-proxy
+  template:
+    metadata:
+      labels:
+        app: submariner-metrics-proxy
+    spec:
+      nodeSelector:
+        "submariner.io/gateway": "true"
+      containers:
+        - name: gateway-metrics-proxy
+          image: quay.io/submariner/nettest:devel
+          imagePullPolicy: IfNotPresent
+          env:
+          - name: NODE_IP
+            valueFrom:
+              fieldRef:
+                fieldPath: status.hostIP
+          - name: SUBM_GW_METRICS_PORT
+            value: "32780"
+          command: ["/usr/bin/nc"]
+          args: ["-v", "-lk", "-p", "8080", "-e", "nc", "$(NODE_IP)", "$(SUBM_GW_METRICS_PORT)"]
+        - name: globalnet-metrics-proxy
+          image: quay.io/submariner/nettest:devel
+          imagePullPolicy: IfNotPresent
+          env:
+          - name: NODE_IP
+            valueFrom:
+              fieldRef:
+                fieldPath: status.hostIP
+          - name: SUBM_GN_METRICS_PORT
+            value: "32781"
+          command: ["/usr/bin/nc"]
+          args: ["-v", "-lk", "-p", "8081", "-e", "nc", "$(NODE_IP)", "$(SUBM_GN_METRICS_PORT)"]
+      tolerations:
+      - operator: Exists
+      restartPolicy: Always
+      schedulerName: default-scheduler
+```
+
+### Changes to Submariner Gateway Metrics service
+
+```diff
+metadata:
+  name: submariner-gateway-metrics
+  labels:
+-   app: submariner-gateway
++   app: submariner-metrics-proxy
+...
+spec:
+  selector:
+-   app: submariner-gateway
++   app: submariner-metrics-proxy
+```
+
+### Changes to Submariner Globalnet Metrics service
+
+```diff
+metadata:
+  name: submariner-globalnet-metrics
+  labels:
+-   app: submariner-globalnet
++   app: submariner-metrics-proxy
+...
+spec:
+  selector:
+-   app: submariner-globalnet
++   app: submariner-metrics-proxy
+```
+
+### Additional Details
+
+These changes will be done in following order to avoid any breakage in CI:
+
+1. Modifying `subctl diagnose` to actually query metrics than check for metrics port availability on firewall.
+2. Create and deploy the new metrics proxy service. As part of this change gateway and globalnet
+daemonset will be modified to set the metrics port in an env variable. It will still use the existing ports `8080`
+and `8081`. This change will be in `submariner-operator` repo.
+3. Modify gateway and globalnet code to use the new metrics port env variable when creating metrics http server. If not
+available, it will default to the new values. This change will be in `submariner` repo.
```

Comment:
```suggestion
3. Modify gateway and globalnet code to use the new metrics port env variable when creating their metrics HTTP servers. If not
available, it will default to the new values. This change will be in the `submariner` repo.
```